### PR TITLE
added resource delete test case without --ids

### PR DESCRIPTION
**Pull request by williexu**
- added `resource delete` case to test command without `--ids`
- changed test_resource_scenario to use ScenarioTest class in the process

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [na ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [na ] Each command and parameter has a meaningful description.
- [na ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py [hunk 1]
```diff
@@ -67,63 +67,47 @@ def body(self):
...
-        s.cmd('resource list --name {}'.format(self.vnet_name),
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
-        s.cmd('resource list --tag cli-test',
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
-        s.cmd('resource list --tag cli-test=test',
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
+            resource_group, vnet_name, subnet_name))
+
+        self.cmd('resource list', checks=JCheck("length([?name=='{}'])".format(vnet_name), 1))
+        self.cmd('resource list -l centralus', checks=JCheck("length([?location == 'centralus']) == length(@)", True))
```
Comment (troydai) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: Can we use `southcentralus` instead? The traffic to that location is lower and is recommended for automation and internal use.

Comment (williexu) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: I was not aware, can def do.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py [hunk 2]
```diff
@@ -67,63 +67,48 @@ def body(self):
...
-    def test_resource_scenario(self):
-        self.execute()
-
-    def __init__(self, test_method):
-        super(ResourceScenarioTest, self).__init__(__file__, test_method,
-                                                   resource_group='cli_test_resource_scenario')
-        self.vnet_name = 'cli-test-vnet1'
-        self.subnet_name = 'cli-test-subnet1'
+class ResourceScenarioTest(ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_rsrc_scenario')
```
Comment (troydai) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: Create the resource group in "southcentralus" as well. And shorten the prefix. If the default prefix works, use the default one.

Comment (williexu) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: errr, @tjprescott recommends prefixes. What to do?

Comment (troydai) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: Travis always won.

Comment (troydai) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: Just make sure the resource group is in southcentralus (a.k.a Texas) then.

Comment (williexu) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: HAHAH okay.

Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py]: There's no harm in NOT using the prefix, but if you ever want to debug into a test as it records and want to inspect the resource group, pure gibberish makes it hard to find in the portal. The prefix makes it much easier. That's the main reason all my tests have one.

**Pull request merged**